### ISSUE TITLE
Turnkey-Snipe-IT

### ISSUE THREAD

**Issue by Montana420** [turnkeylinux/tracker#1870]
Used rufus to image an 8gb usb with the snipe-it iso (https://www.turnkeylinux.org/snipe-it) tested on 6 different PC models and not a single one worked. Also tested with Etcher.

**Comment by JedMeister** [comment]
Hi, thanks for your report.

TL;DR: my guess is that all these are UEFI systems (which TKL does not currently support). My suggestion is to install a hypervisor to bare metal and install TKL as a VM. Read on for more specifics...

----

I'm a Linux user so have limited experience with 3rd party tools to "burn" an ISO to a USB (on Linux you can just copy the ISO direct to the USB via the CLI). Although I have heard that rufus is one that is generally pretty good.

When you say "not a single one worked" that's not really enough info for me to legitimately guess what might be going on. Please provide a bit more info. Please at least answer these questions:

- Was the "bad" behaviour of the "6 different PC models" exactly the same, or were any of them different?
- At what stage did the TurnKey USB "not work"? Did it fail to boot at all? Did it boot ok, but the install failed at a specific step? If so which one? Did it install ok but then something failed during firstboot?
- Were there any error messages displayed at any point? If so what was the explicit text?

Regardless, it sounds like you are trying to install to bare metal. Whilst in theory that should work (and in a perfect world it would), we're a small team with a long list of todos. As TurnKey is primarily intended to run as a VM, we haven't poured tons of resources into improving hardware support and TKL hardware support is not as good as it probably should be. We've made some hardware support improvements in v18 (included non-free firmware by default - although we don't have a v18.0 release for Snipe-IT yet) and have plans for future improvements, but it's not current;y high priority.

If the behaviour was consistent on all "6 different PC models" and the USB failed to boot at all and they are all built within the last 5-10 years and if I was a betting man, I'd bet that they all have UEFI BIOS. And unfortunately, that is a [known issue](https://github.com/turnkeylinux/tracker/issues/1435). We have made some progress on that, but not yet got it resolved. FWIW we have some new ideas If my guess is right and any of the hardware supports disabling UEFI (i.e. enable "legacy BIOS" or similar) then give that a try and fingers crossed it should work (or at least you should hit new and different problems).

Having said all that, unless you expect this server to be under particularly heavy load, or the hardware you're trying to use is particularly low spec, then IMO installing to hardware is a waste of resources and reduces flexibility - e.g. maintenance and troubleshooting is much easier on a VM where you can take snapshots, try things and wind back and/or launch a new server from the snapshot if things don't go to plan. So IMO you'd be better off installing a hypervisor to hardware, then installing TurnKey as a VM on that.

My hypervisor of choice is [Proxmox VE](https://pve.proxmox.com) (disclaimer: I'm biased - TurnKey has a non-financial partnership with Promox). Proxmox is a Debian Linux based (same as TurnKey) hypervisor that is primarily intended to be installed to bare metal, thus has much better hardware support. It also supports 2 types of "virtualization" technology; "normal"/"proper"/"full" VMs (via [KVM](https://en.wikipedia.org/wiki/Kernel-based_Virtual_Machine)/[QEMU](https://en.wikipedia.org/wiki/QEMU) - which can support any OS - inc Windows 11) and [LXC](https://en.wikipedia.org/wiki/LXC) (LinuX Containers - supports Linux only but is extremely low overhead). PVE provides a powerful web UI (so is relatively easy to get started with) and an even more powerful CLI (once you get more familiar with it, you may find that faster and easier?).

As well as ISO, TurnKey is available as LXC (TKL LXC templates are downloadable via the PVE UI). Not all software, workloads and/or purposes work smoothly under LXC (some may need specific config), but most TurnKey appliances should "just work". KVM/QEMU overhead isn't too bad and has better isolation, but given similar resources, you should not notice _any_ performance difference between a guest running in LXC vs bare metal install.

**Comment by OnGle** [comment]
Closing in favor of https://github.com/turnkeylinux/tracker/issues/1435
